src/memory_manager.rs

```diff
@@ -185,6 +185,7 @@ pub fn start_worker<VM: VMBinding>(
 /// * `tls`: The thread that wants to enable the collection. This value will be passed back to the VM in
 ///   Collection::spawn_worker_thread() so that the VM knows the context.
 pub fn enable_collection<VM: VMBinding>(mmtk: &'static MMTK<VM>, tls: VMThread) {
+    mmtk.plan.pre_worker_spawn(mmtk);
```

Comment:
gc_init happens before enable_collection. The number of worker threads in the scheduler is not set up in gc_init. So moving defrag init code to gc_init won't work.

Reply:
My opnion:
1. You don't actually need to get the worker number from the scheduler. You can get it from `Options.threads`. Each plan has a reference to `Options`.
2. I don't mind adding a new hook if it is necessary. E.g. we do want to do something exactly before spawning workers but after gc_init. However, for this case (`initialize_defrag()`), I think it is okay to do it in gc_init.

Reply:
This hook is removed